contracts/FIFOMarket.sol

```diff
@@ -536,6 +511,28 @@ contract FIFOMarket is
     return true; // returns true if the value was added to the set, that is, if it was not already present
   }
 
+  // todo consider making this a generalized `withdrawRemoval`?
+  function release(uint256 removalId, uint256 amount) external returns (bool) {
+    // todo whenNotPaused
+    // todo require sender is Removal address
+    address supplierAddress = removalId.supplierAddress();
+    uint256 removalBalance = _removal.balanceOf(address(this), removalId);
+    if (amount == removalBalance) {
+      require(
+        _activeSupply[supplierAddress].removeRemoval(removalId) ||
+          _reservedSupply.remove(removalId),
+        "Market: removal not in supply"
+      ); // todo custom error
+    }
+    totalReservedSupply += removalBalance; // todo should rely on balanceOf?
```

Comment:
I don't think we want to increment this variable, because releasing a removal is not the same thing as reserving it.  reserving it moves it to a different pool where it's still valid and can be moved back into the active pool later.  releasing a removal means that carbon can never be sold.